ImGuiFileDialog.cpp

```diff
@@ -1,72 +1,100 @@
-#include "ImGuiFileDialog.h"
+// From https://github.com/aiekick/ImGuiFileDialog
+//
+// Modified by AlastairGrowcott@yahoo.com to work on Posix systems.
+
+
+#include <sys/types.h>
+#include <dirent.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <algorithm>
+
+#include "imgui_file_dlg.h"
 #include "imgui.h"
-#include "dirent/dirent.h"
 
 #define IMGUI_DEFINE_MATH_OPERATORS
 #include "imgui_internal.h"
 
-#include <algorithm>
 
-inline bool ReplaceString(std::string& str, const std::string& oldStr, const std::string& newStr)
+#define PATH_SEP '/'
+
+
+static const std::string s_fs_root(1u, PATH_SEP);
+
+
+inline void ReplaceString(std::string& str, const std::string& oldStr, const std::string& newStr)
 {
-	bool Finded = false;
-	size_t pos = 0;
-	while ((pos = str.find(oldStr, pos)) != std::string::npos) {
-		Finded = true;
-		str.replace(pos, oldStr.length(), newStr);
-		pos += newStr.length();
-	}
-	return Finded;
+    size_t pos = 0;
+
+    while ((pos = str.find(oldStr, pos)) != std::string::npos) {
+        str.replace(pos, oldStr.length(), newStr);
+        pos += newStr.length();
+    }
 }
 
 inline std::vector<std::string> splitStringVector(const std::string& text, char delimiter)
 {
-	std::vector<std::string> arr;
-	std::string::size_type start = 0;
-	std::string::size_type end = text.find(delimiter, start);
```

Comment:
Your code here is actually more correct, but much more verbose. I believe that a std::string::size-type is always a size_t in practise, but in theory it doesn't have to be.